src/fidesops/ops/service/connectors/email_connector.py

```diff
@@ -46,5 +48,57 @@ def mask_data(  # type: ignore
         policy: Policy,
         privacy_request: PrivacyRequest,
         rows: List[Row],
+        input_data: Dict[str, List[Any]],
     ) -> Optional[int]:
-        """Stub"""
+        """Cache instructions for how to mask data in this collection.
+        One email will be sent for all collections in this dataset at the end of the privacy request execution.
+        """
+
+        manual_action: Optional[ManualAction] = self.build_masking_instructions(
+            node, policy, input_data
+        )
+
+        if not manual_action:
+            logger.info("No masking required for collection %s", node.address.value)
+            return 0
+
+        privacy_request.cache_email_connector_template_contents(
+            step=CurrentStep.erasure,
+            collection=node.address,
+            action_needed=[manual_action],
+        )
+
+        return 0  # Fidesops itself does not mask this collection.
+
+    def build_masking_instructions(
+        self, node: TraversalNode, policy: Policy, input_data: Dict[str, List[Any]]
+    ) -> Optional[ManualAction]:
+        """
+        Generate information on how to find and mask relevant records on this collection.
+
+        Because we don't have any rows from a completed "access request" for this source, we pass on both "access"
+        and "erasure" instructions to the source.
+        """
+
+        # Values for locating relevant records on the current collection.
+        locators: Dict[str, Any] = node.typed_filtered_values(input_data)
+        # Add additional locators from the same dataset for which we don't have data
+        for edge in node.incoming_edges_from_same_dataset():
+            append(locators, edge.f2.field_path.string_path, str(edge.f1))
+
+        # Build which fields to mask and how to mask
+        fields_to_mask_by_rule: Dict[Rule, List[FieldPath]] = self.query_config(
+            node
+        ).build_rule_target_field_paths(policy)
+        mask_map: Dict[str, Any] = {}
+        for rule, field_paths in fields_to_mask_by_rule.items():
+            for rule_field_path in field_paths:
+                # Map field paths to the specified masking strategy.  If multiple rules target
+                # the same field path, the last one will win
+                mask_map[rule_field_path.string_path] = (
+                    rule.masking_strategy.get("strategy")
+                    if rule.masking_strategy
+                    else None
+                )
+
```

Comment:
I'm caching the masking strategy here, but we will probably just send a list of fields they should mask, not the strategy.  I didn't think it hurt to cache the strategy still in case that is useful later.

Reply:
This is good — I could also imagine us precomputing values for the third party to update PII to also